cinema/tests/test_movie_viewset.py

```diff
@@ -0,0 +1,162 @@
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
+
+    def test_filter_movies_by_title(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"title": "Movie One"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_filter_movies_by_genres(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"genres": f"{self.genre1.id}"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_filter_movies_by_actors(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"actors": f"{self.actor1.id}"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_create_movie_as_authenticated_user(self):
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+        response = self.client.post(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        expected_error = {
+            "detail": ErrorDetail(
+            string="You do not have permission to perform this action.",
+            code="permission_denied"
+            )
+        }
+        self.assertEqual(response.data, expected_error)
+
+    def test_create_movie_as_admin(self):
+        self.client.credentials(HTTP_AUTHORIZATION=f"Token {self.admin_token.key}")
+
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+
+        response = self.client.post(url, data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(response.data["title"], "New Movie")
+
+        movie = Movie.objects.get(title="New Movie")
+
+        self.assertEqual(movie.description, "A new movie description")
+        self.assertEqual(movie.duration, 90)
+        self.assertEqual(list(movie.genres.values_list('id', flat=True)), [self.genre1.id])
+        self.assertEqual(list(movie.actors.values_list('id', flat=True)), [self.actor1.id, self.actor2.id])
+
+    def test_retrieve_movie(self):
```

Comment:
This test checks if a movie can be retrieved. It's a good practice to also check if the retrieved movie data matches the expected data. You're doing this for the title, genres, and actors, which is good. However, consider also checking the other fields, like the description and duration, to ensure that all the movie data is correctly retrieved.